tests/Http/AcceptHeaderTest.php

```diff
@@ -0,0 +1,163 @@
+        $this->assertEquals(1, $compareValue->invokeArgs(null, ['type/*', '*/*']));
+        $this->assertEquals(1, $compareValue->invokeArgs(null, ['type/subtype', '*/*']));
+        $this->assertEquals(-1, $compareValue->invokeArgs(null, ['*/*', 'type/*']));
+        $this->assertEquals(-1, $compareValue->invokeArgs(null, ['*/*', 'type/subtype']));
+
+        $this->assertEquals(0, $compareValue->invokeArgs(null, ['*/*', '*/*']));
+        $this->assertEquals(0, $compareValue->invokeArgs(null, ['type/*', 'type/*']));
+        $this->assertEquals(1, $compareValue->invokeArgs(null, ['type/subtype', 'type/*']));
+        $this->assertEquals(-1, $compareValue->invokeArgs(null, ['type/*', 'type/subtype']));
+    }
+
+    public function testHeaderItem()
+    {
+        $item = new AcceptHeaderItem('text/html', 0.9, ['t' => 'test']);
+
+        $this->assertEquals('text/html; q=0.9; t=test', $item);
+        $this->assertEquals(0.9, $item->getQuality());
+        $this->assertEquals(['t' => 'test'], $item->getParams());
+
+        $item = $item->withValue('application/json');
+
+        $this->assertEquals('application/json', $item->getValue());
+
+        $item = $item->withQuality(0);
+
+        $this->assertEquals(0, $item->getQuality());
+
+        $item = $item->withParams(['n' => 'new']);
+
+        $this->assertEquals(['n' => 'new'], $item->getParams());
+
+        $wrongParamsItem = AcceptHeaderItem::fromString('text/html; q');
+
+        $this->assertEquals([], $wrongParamsItem->getParams());
+        $this->assertEquals(1.0, $wrongParamsItem->getQuality());
+
+        AcceptHeaderItem::fromString('');
+    }
+
+    public function testHeaderConstructing()
+    {
+        $acceptCharset = new AcceptHeader(['*', 'UTF-8']);
+
+        $this->assertFalse($acceptCharset->has('unicode'));
+        $this->assertNull($acceptCharset->get('unicode'));
+
+        $sorted = $acceptCharset->sorted();
+        $this->assertEquals('UTF-8', $sorted[0]);
+        $this->assertEquals('*', $sorted[1]);
+        $this->assertEquals('UTF-8, *', $acceptCharset);
+
+        $acceptCharset = $acceptCharset->add(AcceptHeaderItem::fromString('unicode'));
+
+        $this->assertTrue($acceptCharset->has('unicode'));
+        $this->assertNotNull($acceptCharset->get('unicode'));
+
+        $sorted = $acceptCharset->sorted();
+        $this->assertEquals('UTF-8', $sorted[0]);
+        $this->assertEquals('unicode', $sorted[1]);
+        $this->assertEquals('*', $sorted[2]);
+        $this->assertEquals('UTF-8, unicode, *', $acceptCharset);
+
+        $acceptHeader = AcceptHeader::fromString('text/html; q=0.8, */*; q=0.7');
+        $acceptHeader = $acceptHeader->add('application/json');
+
+        $sorted = $acceptHeader->sorted();
+        $this->assertEquals('application/json', $sorted[0]);
+        $this->assertEquals('text/html; q=0.8', $sorted[1]);
+        $this->assertEquals('*/*; q=0.7', $sorted[2]);
+    }
+
```

Comment:
What will happen if the accept header will receive an item of unsupported type (neither `AcceptHeaderItem` nor `string`)? The next tese case will fail
```suggestion

    public function testHeaderConstructingTypeError(): void
    {
        $acceptCharset = new AcceptHeader(['*', 'UTF-8', ['text/html;level=1']]);

        self::assertCount(2, $acceptCharset->all());
    }
```